plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/clients/KafkaClientsModule.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.kafka.clients;
+
+import com.google.inject.Binder;
+import com.google.inject.Module;
+import com.google.inject.Scopes;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.trino.plugin.kafka.KafkaConfig;
+import io.trino.plugin.kafka.clients.admin.KafkaAdminFactory;
+import io.trino.plugin.kafka.clients.admin.PlainTextKafkaAdminFactory;
+import io.trino.plugin.kafka.clients.admin.SslKafkaAdminFactory;
+import io.trino.plugin.kafka.clients.consumer.KafkaConsumerFactory;
+import io.trino.plugin.kafka.clients.consumer.PlainTextKafkaConsumerFactory;
+import io.trino.plugin.kafka.clients.consumer.SslKafkaConsumerFactory;
+import io.trino.plugin.kafka.clients.producer.KafkaProducerFactory;
+import io.trino.plugin.kafka.clients.producer.PlainTextKafkaProducerFactory;
+import io.trino.plugin.kafka.clients.producer.SslKafkaProducerFactory;
+import io.trino.plugin.kafka.security.ForKafkaSecurity;
+import io.trino.plugin.kafka.security.SecurityProtocol;
+
+import static io.airlift.configuration.ConditionalModule.installModuleIf;
+
+public class KafkaClientsModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        installClientModule(SecurityProtocol.PLAINTEXT, KafkaClientsModule::configurePlainText);
+        installClientModule(SecurityProtocol.SSL, KafkaClientsModule::configureSsl);
+    }
+
+    private void installClientModule(SecurityProtocol securityProtocol, Module module)
+    {
+        install(installModuleIf(
+                KafkaConfig.class,
+                config -> config.getSecurityProtocol().equals(securityProtocol),
+                module));
+    }
+
+    private static void configurePlainText(Binder binder)
+    {
+        binder.bind(KafkaConsumerFactory.class).to(PlainTextKafkaConsumerFactory.class).in(Scopes.SINGLETON);
+        binder.bind(KafkaProducerFactory.class).to(PlainTextKafkaProducerFactory.class).in(Scopes.SINGLETON);
+        binder.bind(KafkaAdminFactory.class).to(PlainTextKafkaAdminFactory.class).in(Scopes.SINGLETON);
+    }
+
+    private static void configureSsl(Binder binder)
+    {
+        binder.bind(KafkaConsumerFactory.class).annotatedWith(ForKafkaSecurity.class).to(PlainTextKafkaConsumerFactory.class).in(Scopes.SINGLETON);
```

Comment:
Use `ForKafkaSsl`. Security is a very vast term. It can either mean security protocol or authentication mechanism.